### PR TITLE
57 table 4@main

### LINKED ISSUE #57: Table 4 - Patient Disposition, Pooled Analyses
## Table Layout & Design

table04

table04-2

Please refer to page 14-15 of the pdf here for more details: [FDA-2022-N-1961-0002_attachment_1.pdf](https://github.com/pharmaverse/falcon/files/10744259/FDA-2022-N-1961-0002_attachment_1.pdf)

## Description

Patient Disposition, Pooled Analyses

## Analysis Datasets

Any dataset - df
**ADSL** - alt_counts_df (if required)

When using data from the `scda` package, please use datasets with `archive_name` set to `"rcd_2022_10_13"`.

## Analysis Filtering

table04-3

## Column Variable

**ARM** (ignore "Risk Difference" column for now. This will be added in at a later date once functionality is available.)

## Additional Notes

N/A

### PR DESCRIPTION
**Pull request by Aura1201**
I have created the function for table 4 and made some changes to the former version. Please check and review.

Closes #57

### CODE REVIEW

File: R/fda-table_04.R [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+#' * Flag variables (i.e. `XXXFL`) are expected to have two levels: `"Y"` (true) and `"N"` (false). Missing values in
+#'   flag variables are treated as `"N"`.
+#' * Columns are split by arm. Overall population column is excluded by default (see `lbl_overall` argument).
+#' * Numbers in table "Patients Treated" section are the absolute numbers of patients and fraction of `N`.
+#' * All-zero rows are not removed by default (see `prune_0` argument).
+#' * Records with missing treatment start and/or end datetime are excluded from all calculations.
+#'
+#' @inheritParams argument_convention
+
+#' @examples
```
Comment (jknizia) [inline R/fda-table_04.R:16]: Could you please double check whether this example works on your end? I tried to run it, but it throws an error:
image

**Review by jknizia — CHANGES_REQUESTED** [review]
Hi @Aura1201, 
thank you for adding the table. The table generation seems to work in general, however, the example you provided in the roxygen documentation does not work (see my comment in the file.) Please double check this issue :-)

File: R/fda-table_04.R (hunk 1 above)
Comment (jknizia) [reply R/fda-table_04.R:16]: FYI, I withdraw my assignment to act as a reviewer since I will be ooo until July 24th.

Comment (Aura1201) [reply R/fda-table_04.R:16]: Hi Jessica, I have updated the function for table 4 and it seems there is no errors like that. Thank you

Comment (jknizia) [reply R/fda-table_04.R:16]: It is only the example in the roxygen documentation that does not work. The problem is that your example adsl does not have columns "RANDFL", "PPROTFL", and "DCTREAS", which causes the function to stop. So if you provide the code to create dummy columns for these variables, it should work.

Comment (Aura1201) [reply R/fda-table_04.R]: Hi Jessica, the new fda-table_04 has been updated to our 57 branch, please take a look when available.

**Review by jknizia — APPROVED** [review]
Hi Yuye, looks great!

**Comment by Aura1201** [comment]
> Hi Yuye, looks great!

Hi Jessica, I notice that our table 4 has not been merged into main branch yet. Is there anything need to update for table 4? If yes, please feel free to make comments or contact me in Slack. Thank you.

**Comment by jknizia** [comment]
Hi Yuye,
everything good from my side. I have already approved the PR some time ago. Please feel free to merge anytime! :-)

**Pull request merged**